File: app/src/main/java/xyz/volgoak/wordlearning/recycler/ChoiceMode.java
package xyz.volgoak.wordlearning.recycler;

import android.os.Bundle;

import java.util.List;

/**
 * Created by Volgoak on 20.08.2017.
 */

public interface ChoiceMode {
    void setChecked(int position, boolean checked);
    boolean isChecked(int position);
    void onSaveInstanceState(Bundle instanceState);
    void restoreInstanceState(Bundle instanceState);
    int getCheckedCount();
    void clearChecks();
    int getCheckedPosition();
    List<Integer> getCheckedList();
}

File: app/src/main/java/xyz/volgoak/wordlearning/adapter/choice_mode/ChoiceMode.kt
package xyz.volgoak.wordlearning.adapter.choice_mode

import android.os.Bundle

/**
 * Created by Volgoak on 20.08.2017.
 */

interface ChoiceMode {
    val checkedCount: Int
    val checkedPosition: Int
    val checkedList: List<Int>
    fun setChecked(position: Int, checked: Boolean)
    fun isChecked(position: Int): Boolean
    fun onSaveInstanceState(instanceState: Bundle)
    fun restoreInstanceState(instanceState: Bundle)
    fun clearChecks()
}
